### PR TITLE
Clean up remote attestation tests

### PR DESCRIPTION
**Pull request by clangenb**
Preliminary clean up before I actually implement the DCAP benchmarks.

Changes:
* Removed duplicate definitions of `consts` based on test data binaries. Now, all the test-data constants are **defined** in the `sgx-verify` crate and simply **re-exported** in the `test_utils` crate.
* Removed unused test-data definitions and fixed compiler warnings of these unused constants

### CODE REVIEW

File: teerex/sgx-verify/src/tests.rs [hunk 1]
```diff
@@ -1,129 +1,60 @@
...
+const QE_IDENTITY_CERT: &str = include_str!("../test-data/dcap/qe_identity_cert.pem");
+const DCAP_QUOTE_CERT: &str = include_str!("../test-data/dcap/dcap_quote_cert.der");
+const PCK_CRL: &[u8] = include_bytes!("../test-data/dcap/pck_crl.der");
 
-const TEST1_SIGNER_ATTN: &[u8] =
-	include_bytes!("../test/test_ra_signer_attn_MRSIGNER1_MRENCLAVE1.bin");
-const TEST2_SIGNER_ATTN: &[u8] =
-	include_bytes!("../test/test_ra_signer_attn_MRSIGNER2_MRENCLAVE2.bin");
-const TEST3_SIGNER_ATTN: &[u8] =
-	include_bytes!("../test/test_ra_signer_attn_MRSIGNER3_MRENCLAVE2.bin");
```
Comment (clangenb) [inline teerex/sgx-verify/src/tests.rs:22]: These were unused.

File: teerex/sgx-verify/src/tests.rs [hunk 2]
```diff
@@ -1,129 +1,60 @@
...
-
-// reproduce with "integritee_service signing-key"
-const TEST1_SIGNER_PUB: &[u8] =
-	include_bytes!("../test/test_ra_signer_pubkey_MRSIGNER1_MRENCLAVE1.bin");
-const TEST2_SIGNER_PUB: &[u8] =
-	include_bytes!("../test/test_ra_signer_pubkey_MRSIGNER2_MRENCLAVE2.bin");
-const TEST3_SIGNER_PUB: &[u8] =
-	include_bytes!("../test/test_ra_signer_pubkey_MRSIGNER3_MRENCLAVE2.bin");
```
Comment (clangenb) [inline teerex/sgx-verify/src/tests.rs:30]: Unused...

File: teerex/sgx-verify/src/tests.rs [hunk 3]
```diff
@@ -1,129 +1,60 @@
...
-	138, 157, 86, 136, 77, 91, 93, 23, 192, 104, 140, 167,
-];
-const TEST2_MRENCLAVE: &[u8] = &[
-	4, 190, 230, 132, 211, 129, 59, 237, 101, 78, 55, 174, 144, 177, 91, 134, 1, 240, 27, 174, 81,
-	139, 8, 22, 32, 241, 228, 103, 189, 43, 44, 102,
-];
-const TEST3_MRENCLAVE: &[u8] = &[
-	4, 190, 230, 132, 211, 129, 59, 237, 101, 78, 55, 174, 144, 177, 91, 134, 1, 240, 27, 174, 81,
-	139, 8, 22, 32, 241, 228, 103, 189, 43, 44, 102,
-];
```
Comment (clangenb) [inline teerex/sgx-verify/src/tests.rs:52]: unused...

File: teerex/sgx-verify/src/tests.rs [hunk 4]
```diff
@@ -1,129 +1,60 @@
...
...cQjlytKgN9cLnxbwtuvLUK7eyRPfJW/ksddOzP8VBBniolYnRCD2jrMRZ8nBM2ZWYwnXnwYeOAHV+W9tOhAImwRwKF/95yAsVwd21ryHMJBcGH70qLagZ7Ttyt++qO/6+KAXJuKwZqjRlEtSEz8gZQeFfVYgcwSfo96oSMAzVr7V0L6HSDLRnpb6xxmbPdqNol4tQIDAQABo4GkMIGhMB8GA1UdIwQYMBaAFHhDe3amfrzQr35CN+s1fDuHAVE8MA4GA1UdDwEB/wQEAwIGwDAMBgNVHRMBAf8EAjAAMGAGA1UdHwRZMFcwVaBToFGGT2h0dHA6Ly90cnVzdGVkc2VydmljZXMuaW50ZWwuY29tL2NvbnRlbnQvQ1JML1NHWC9BdHRlc3RhdGlvblJlcG9ydFNpZ25pbmdDQS5jcmwwDQYJKoZIhvcNAQELBQADggGBAGcIthtcK9IVRz4rRq+ZKE+7k50/OxUsmW8aavOzKb0iCx07YQ9rzi5nU73tME2yGRLzhSViFs/LpFa9lpQL6JL1aQwmDR74TxYGBAIi5f4I5TJoCCEqRHz91kpG6Uvyn2tLmnIdJbPE4vYvWLrtXXfFBSSPD4Afn7+3/XUggAlc7oCTizOfbbtOFlYA4g5KcYgS1J2ZAeMQqbUdZseZCcaZZZn65tdqee8UXZlDvx0+NdO0LR+5pFy+juM0wWbu59MvzcmTXbjsi7HY6zd53Yq5K244fwFHRQ8eOB0IWB+4PfM7FeAApZvlfqlKOlLcZL2uyVmzRkyR5yW72uo9mehX44CiPJ2fse9Y6eQtcfEhMPkmHXI01sN+KwPbpA39+xOsStjhP9N1Y1a2tQAVo+yVgLgV2Hws73Fc0o3wC78qPEA+v2aRs/Be3ZFDgDyghc/1fgU+7C+P6kbqd4poyb6IW8KCJbxfMJvkordNOgOUUxndPHEi/tb/U7uLjLOgPA==0\n\x06\x08*\x86H\xce=\x04\x03\x02\x03H\x000E\x02!\0\xae6\x06\t@Sy\x8f\x8ec\x9d\xdci^Ex*\x92}\xdcG\x15A\x97\xd7\xd7\xd1\xccx\xe0\x1e\x08\x02 \x15Q\xa0BT\xde'~\xec\xbd\x027\xd3\xd8\x83\xf7\xe6Z\xc5H\xb4D\xf7\xe2\r\xa7\xe4^f\x10\x85p";
```
Comment (clangenb) [inline teerex/sgx-verify/src/tests.rs:77]: Inline that to the actual test that uses it.

File: teerex/sgx-verify/src/tests.rs [hunk 5]
```diff
@@ -1,129 +1,60 @@
...
-
 #[test]
 fn verify_ias_report_should_work() {
-	let _signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
 	let report = verify_ias_report(TEST4_CERT);
 	let report = report.unwrap();
 	assert_eq!(report.mr_enclave, TEST4_MRENCLAVE);
-	assert!(report.timestamp >= TEST1_TIMESTAMP.try_into().unwrap());
-	assert_eq!(report.pubkey, TEST4_SIGNER_PUB);
+	assert_eq!(report.timestamp, TEST4_TIMESTAMP);
```
Comment (clangenb) [inline teerex/sgx-verify/src/tests.rs:24]: I did not understand what `report.timestamp >= TEST1_TIMESTAMP.try_into().unwrap()` was supposed to test. I think it is more accurate to test that the timestamp actually matches.

File: teerex/sgx-verify/src/tests.rs [hunk 6]
```diff
@@ -1,129 +1,60 @@
...
+	assert_eq!(&report.pubkey, TEST4_SIGNER_PUB);
 	//assert_eq!(report.status, SgxStatus::GroupOutOfDate);
 	assert_eq!(report.status, SgxStatus::ConfigurationNeeded);
 	assert_eq!(report.build_mode, SgxBuildMode::Debug);
 }
 
 #[test]
 fn verify_zero_length_cert_returns_err() {
-	// CERT empty, argument 2 and 3 are wrong too!
-	let _signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
```
Comment (clangenb) [inline teerex/sgx-verify/src/tests.rs:99]: dafuq, who had put this everywhere?

File: teerex/sgx-verify/src/tests.rs [hunk 7]
```diff
@@ -1,129 +1,60 @@
...
+	},
+};
 use codec::Decode;
 use frame_support::assert_err;
 use hex_literal::hex;
 
-// reproduce with "integritee_service dump_ra"
-const TEST1_CERT: &[u8] = include_bytes!("../test/test_ra_cert_MRSIGNER1_MRENCLAVE1.der");
-const TEST2_CERT: &[u8] = include_bytes!("../test/test_ra_cert_MRSIGNER2_MRENCLAVE2.der");
-const TEST3_CERT: &[u8] = include_bytes!("../test/test_ra_cert_MRSIGNER3_MRENCLAVE2.der");
```
[the remark this answers is not in the document]
Comment (clangenb) [reply teerex/sgx-verify/src/tests.rs:10]: I was not the one who implemented these tests. @brenzi, @haerdib can you please briefly comment if there was a concrete plan with this?

[the remark this answers is not in the document]
Comment (haerdib) [reply teerex/sgx-verify/src/tests.rs:10]: Happened before my time, so I can't help you there. Check out https://github.com/integritee-network/pallet-teerex/blame/32eec55c0b4691390745bbc27ddc6bd588d04c05/ias-verify/src/lib.rs if you want to know

[the remark this answers is not in the document]
Comment (brenzi) [reply teerex/sgx-verify/src/tests.rs:10]: my time. but I don't remember why they aren't used anymore or what they were used for

[the remark this answers is not in the document]
Comment (brenzi) [reply teerex/sgx-verify/src/tests.rs:10]: I think they were just a set of edge cases, so I hope we still cover them with our tests

[the remark this answers is not in the document]
Comment (brenzi) [reply teerex/sgx-verify/src/tests.rs:10]: me be worth investigating TBH

[the remark this answers is not in the document]
Comment (clangenb) [reply teerex/sgx-verify/src/tests.rs:10]: Out of scope for this task. I filed an issue: https://github.com/integritee-network/pallets/issues/169

File: teerex/src/lib.rs [hunk 8]
```diff
@@ -551,8 +546,8 @@ impl<T: Config> Pallet<T> {
 	fn verify_report(
 		sender: &T::AccountId,
 		ra_report: Vec<u8>,
-	) -> Result<SgxReport, DispatchErrorWithPostInfo> {
-		let report = verify_ias_report(&ra_report)
+	) -> Result<sgx_verify::SgxReport, DispatchErrorWithPostInfo> {
+		let report = sgx_verify::verify_ias_report(&ra_report)
 			.map_err(|_| <Error<T>>::RemoteAttestationVerificationFailed)?;
```
Comment (clangenb) [inline teerex/src/lib.rs:551]: There were unused imports with the `skip-ias-check` feature flag. Instead of adding a `#[cfg(feature ..)]` I inlined the imports as I found it more handy.

File: teerex/src/weights.rs [hunk 9]
```diff
@@ -67,13 +67,11 @@ impl<T: frame_system::Config> WeightInfo for IntegriteeWeight<T> {
 			.saturating_add(T::DbWeight::get().reads(2))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
-	// TODO benchmark dcap registration
```
Comment (clangenb) [inline teerex/src/weights.rs:70]: Oups, accidentally mistake from cherry-pick, but the next PR will resolve it anyhow.

**Review by OverOrion — APPROVED** [review]
Thanks for the clean up, LGTM!
